src/unit_tests/wazuh_db/test_create_agent_db.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2015, Wazuh Inc.
+ * May 2, 2024.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "../wazuh_db/wdb.h"
+#include "../wrappers/wazuh/shared/debug_op_wrappers.h"
+#include "../wrappers/wazuh/shared/file_op_wrappers.h"
+#include "../wrappers/libc/stdio_wrappers.h"
+
+void test_wdb_create_agent_db2_ok(void ** state) {
+    expect_wfopen(WDB2_DIR "/" WDB_PROF_NAME, "r", (void *)1);
+    expect_wfopen(WDB2_DIR "/000.db.new", "w", (void *)2);
+    expect_fread("", 0);
+    expect_fclose((void *)1, 0);
+    expect_fclose((void *)2, 0);
+    expect_string(__wrap_chmod, path, WDB2_DIR "/000.db.new");
+    will_return(__wrap_chmod, 0);
+    expect_string(__wrap_OS_MoveFile, src, WDB2_DIR "/000.db.new");
+    expect_string(__wrap_OS_MoveFile, dst, WDB2_DIR "/000.db");
+    will_return(__wrap_OS_MoveFile, 0);
+
+    int result = wdb_create_agent_db2("000");
+    assert_int_equal(result, 0);
+}
+
+void test_wdb_create_agent_db2_wfopen_error(void ** state) {
+    expect_wfopen(WDB2_DIR "/" WDB_PROF_NAME, "r", (void *)1);
+    expect_wfopen(WDB2_DIR "/000.db.new", "w", NULL);
+    expect_string(__wrap__merror, formatted_msg, "Couldn't create database 'queue/db/000.db': Success (0)");
+    expect_fclose((void *)1, 0);
+
+    errno = 0;
+    int result = wdb_create_agent_db2("000");
+    assert_int_equal(result, -1);
+}
+
+void test_wdb_create_agent_db2_fwrite_error(void ** state) {
+    expect_wfopen(WDB2_DIR "/" WDB_PROF_NAME, "r", (void *)1);
+    expect_wfopen(WDB2_DIR "/000.db.new", "w", (void *)2);
+    expect_fread("Hello", 5);
+    will_return(__wrap_fwrite, 0);
+    expect_fclose((void *)1, 0);
+    expect_fclose((void *)2, 0);
+
+    int result = wdb_create_agent_db2("000");
+    assert_int_equal(result, -1);
+}
+
+void test_wdb_create_agent_db2_fclose_error(void ** state) {
+    expect_wfopen(WDB2_DIR "/" WDB_PROF_NAME, "r", (void *)1);
+    expect_wfopen(WDB2_DIR "/000.db.new", "w", (void *)2);
+    expect_fread("", 0);
+    expect_fclose((void *)1, 0);
+    expect_fclose((void *)2, -1);
+    expect_string(__wrap__merror, formatted_msg, "Couldn't create file queue/db/000.db.new completely");
+
+    int result = wdb_create_agent_db2("000");
+    assert_int_equal(result, -1);
+}
+
```

Comment:
if the above comments sounds correct, we can also add the following test to prove it:

```
void test_wdb_create_agent_db2_fwrite_error_2(void ** state) {
    expect_wfopen(WDB2_DIR "/" WDB_PROF_NAME, "r", (void *)1);
    expect_wfopen(WDB2_DIR "/000.db.new", "w", (void *)2);
    expect_fread("Hello", 5);
    will_return(__wrap_fwrite, 5);
    expect_fclose((void *)1, 0);
    expect_fclose((void *)2, 0);
    expect_string(__wrap__merror, formatted_msg, "Couldn't write succesfully to file queue/db/000.db.new completely");
    
    int result = wdb_create_agent_db2("000");
    assert_int_equal(result, -1);
}
```